compatibility/src/test/java/io/cucumber/compatibility/CompatibilityTest.java

```diff
@@ -114,7 +114,8 @@ void produces_expected_output_for(TestCase testCase) throws IOException {
     }
 
     private void sortStepDefinitions(Map<String, List<JsonNode>> envelopes) {
-        Comparator<JsonNode> stepDefinitionPatternComparator = Comparator.comparing(a -> a.get("pattern").asText());
+        Comparator<JsonNode> stepDefinitionPatternComparator = Comparator
+                .comparing(a -> a.get("pattern").get("source").asText());
```

Comment:
I think the order of the nested json fields was not always the same, causing the comparison order to get reversed

Reply:
Yeah. Order isn't guaranteed with json. Nicely spotted.